popde/density_estimate.py

```diff
@@ -1,3 +1,83 @@
+class SimpleKernelDensityEstimation:
+    """
+    Kernel Density Estimation (KDE) using Kernels
+
+    Methods:
+    --------
+    fit(data, bandwidth=None, kernel=None):
+        Fit the KDE to new data.
+
+    evaluate(points):
+        Evaluate the KDE at given data points.
+
+    Examples:
+    ---------
+    # Create a KDE instance and fit it to data
+    data = np.random.randn(100)
+    kde = SimpleGaussianKernelDensityEstimation(data)
+    kde.fit()
+
+    # Evaluate the KDE at new data points
+    new_data = np.linspace(-3, 3, 100)
+    density_values = kde.evaluate(new_data)
+    """
+
+    def __init__(self, data, bandwidth=1.0, kernel=['gaussian']):
+        """
+        Initialize the KernelDensityEstimation object.
+
+        Parameters:
+        -----------
+        data : array-like
```

Comment:
Can this be multi-dimensional?  I think we want all the classes and methods to be multidimensional from the beginning otherwise there is no point writing new code.

For 1d KDE we already have a complete solution in awkde, the point of having this new package is to deal with issues where we have 2 / 3 / 4 different dimensions in population analysis.

So can you build in the use of multi dimensions from the beginning

Reply:
ok